src/schema/v2/artwork/updateArtworkMutation.ts

```diff
@@ -254,15 +254,14 @@ export const updateArtworkMutation = mutationWithClientMutationId<
           return new Error("You need to be signed in to perform this action")
         }
 
-        // Attach all images
-        await Promise.all(
-          imageS3Locations.map((location) =>
-            addImageToArtworkLoader(id, {
-              source_bucket: location.bucket,
-              source_key: location.key,
-            })
-          )
-        )
+        // Attach all images sequentially to avoid race conditions
+        await imageS3Locations.reduce(async (previousPromise, location) => {
+          await previousPromise
+          return addImageToArtworkLoader(id, {
+            source_bucket: location.bucket,
+            source_key: location.key,
+          })
+        }, Promise.resolve())
```

Comment:
You could also use a simple for loop:

```suggestion
        // Attach all images sequentially
        for (const location of imageS3Locations) {
          await addImageToArtworkLoader(id, {
            source_bucket: location.bucket,
            source_key: location.key,
          })
      }
```

and catch errors separately in case this is helpful 🤷 

```
const errors = [];
for (const loc of imageS3Locations) {
    try {
        await addImageToArtworkLoader(id, {
            source_bucket: loc.bucket,
            source_key: loc.key,
        });
    } catch (err) {
        errors.push({ loc, err });
    }
}
if (errors.length) {
    // handle errors
}
```